riotwatcher/_apis/test/test_BaseApi.py

```diff
@@ -0,0 +1,96 @@
+import unittest
+import sys
+
+if sys.version_info > (3, 0):
+    from unittest.mock import MagicMock
+else:
+    from mock import MagicMock
+from .. import BaseApi
+
+
+class BaseApiTestCase(unittest.TestCase):
+    def setUp(self):
+        self._expected_return = object()
+
+        self._request_handler_mock = MagicMock(name='request_handler')
+        self._request_handler_mock.preview_request = MagicMock(name='preview_request')
+        self._request_handler_mock.preview_request.return_value = self._expected_return
+        self._request_handler_mock.after_request = MagicMock(name='after_request')
+        self._request_handler_mock.after_request.return_value = self._expected_return
+        self._request_handler_mock.preview_static_request = MagicMock(name='preview_static_request')
+        self._request_handler_mock.preview_static_request.return_value = self._expected_return
+        self._request_handler_mock.after_static_request = MagicMock(name='after_static_request')
+        self._request_handler_mock.after_static_request.return_value = self._expected_return
+
+        self._api_key = 'sadf'
+
+    def test_base_api_request(self):
```

Comment:
This is actually testing 2 different behaviors (and can test a 3rd):

1. The `preview_request` handler is called with the correct arguments
2. If a `preview_request` call returns a value, that value is passed to the first `after_request` handler

These 2 test cases should be split into different functions.